community/modules/scheduler/schedmd-slurm-gcp-v5-controller/variables.tf

```diff
@@ -0,0 +1,488 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+###########
+# GENERAL #
+###########
+
+variable "project_id" {
+  type        = string
+  description = "Project ID to create resources in."
+}
+
+variable "slurm_cluster_name" {
+  type        = string
+  description = "Cluster name, used for resource naming and slurm accounting."
+
+  validation {
+    condition     = can(regex("(^[a-z][a-z0-9]*$)", var.slurm_cluster_name))
```

Comment:
it seems to me that if you are going to do any validation, that you should match the [exact regex used by SchedMD](https://github.com/SchedMD/slurm-gcp/blob/master/terraform/slurm_cluster/modules/slurm_partition/variables.tf#L26). I find it a bit odd that they do not allow dashes, but it's fine and they are imposing a 10-character limit on the whole string that you are not.

You should be aware that Terraform will perform the regex check in the wrapper and in the other module so the value here isn't enormous relative to the potential for a conflict.

I also would have the error message say something in "English" rather than "regex". e.g. "the cluster name can be no longer than 10 lowercase alphanumeric characters and must begin with a letter."

Reply:
That's a good point, I removed the validation for this and other passthrough variables to avoid duplication of effort (both for us and for terraform).